pkg/build/builder/source_test.go

```diff
@@ -0,0 +1,29 @@
+package builder
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestCheckRemoteGit(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusUnauthorized)
+	}))
+	defer server.Close()
+
+	var err error
+	err = checkRemoteGit(server.URL, 10*time.Second)
+	switch v := err.(type) {
+	case gitAuthError:
+	default:
+		t.Errorf("expected gitAuthError, got %q", v)
+	}
+	err = checkRemoteGit("https://254.254.254.254/foo/bar", 4*time.Second)
+	if err == nil || (err != nil && !strings.Contains(fmt.Sprintf("%s", err), "timeout")) {
+		t.Errorf("expected timeout error, got %q", err)
+	}
```

Comment:
@liggitt and i were discussing this and it seems like you still need 2 more tests:

1) a valid test that passes to show you can successfully check a remote git (you had this before but you removed it to avoid using github)

2) a test that confirms the actual timeout that checkRemoteGit takes as a parameter, is respected in cases where the git server takes forever to respond.